### ISSUE TITLE
Edge agent cannot start with error "Insufficient system resources exist to complete the requested service."

### ISSUE THREAD

**Issue by Gimly** [Azure/iotedge#3680]
## Expected Behavior
Edge agent should start successfully on a Windows Server 2019 Datacenter 1809 virtualized with 8GB of RAM and dual core 2.2GHz processor.

## Current Behavior
Edge agent does not start, when the daemon tries to start the Docker image it gets an error when trying to run `CreateComputeSystem` with the error `Insufficient system resources exist to complete the requested service.`.

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. Install IoT Edge agent on a virtualized Windows Server 2019 DataCenter 1809 (Build 17763.1457) with the updates KB4570333 and KB4570332 following the explanations at https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge-windows
2. At the end of the installation and initialization try to run the iotedge check command

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```

PS P:\> iotedge check
Configuration checks
--------------------
√ config.yaml is well-formed - OK
√ config.yaml has well-formed connection string - OK
√ container engine is installed and functional - OK
√ Windows host version is supported - OK
√ config.yaml has correct hostname - OK
× config.yaml has correct URIs for daemon mgmt endpoint - Error
    Unable to find image 'mcr.microsoft.com/azureiotedge-diagnostics:1.0.9.5' locally
    1.0.9.5: Pulling from azureiotedge-diagnostics
    6a5e981230da: Pulling fs layer
    f7de0eb29689: Pulling fs layer
    b9472b5cda4b: Pulling fs layer
    b9472b5cda4b: Verifying Checksum
    b9472b5cda4b: Download complete
    6a5e981230da: Verifying Checksum
    6a5e981230da: Download complete
    f7de0eb29689: Verifying Checksum
    f7de0eb29689: Download complete
    6a5e981230da: Pull complete
    f7de0eb29689: Pull complete
    b9472b5cda4b: Pull complete
    Digest: sha256:b56d76881cad70d73106099510acdfb32f78b62e109d62996c4a71a53e3d3f7c
    Status: Downloaded newer image for mcr.microsoft.com/azureiotedge-diagnostics:1.0.9.5
    docker: Error response from daemon: hcsshim::CreateComputeSystem d1e446a8b63c01dee5da6bfc3cf4e1e67258fba5f850cbff22fd5e28bb39115d: Insufficient system resources exist to complete the requested service.
    (extra info: {"SystemType":"Container","Name":"d1e446a8b63c01dee5da6bfc3cf4e1e67258fba5f850cbff22fd5e28bb39115d","Owner":"docker","VolumePath":"\\\\?\\Volume{7acefee5-2633-413c-bd11-dd1f37fc09ca}","IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\d1e446a8b63c01dee5da6bfc3cf4e1e67258fba5f850cbff22fd5e28bb39115d","Layers":[{"ID":"1fba2e47-c004-5a7d-8764-d8a42b59dfe1","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\95283b53a8b32d1088ac51fa06caa159b706d6778d3cfe5d045b56576b02c85b"},{"ID":"fa14c24e-51b7-5e54-9c23-ddc1fa1df49e","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\7761baa93e1f611b78d9d14bf0bee05c54c10bbffab00bfa309c571bfb4aa072"},{"ID":"b5428c02-c8fe-5bb0-b469-77eb2965a34d","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\ea1789788090b5634d53bf3e3615c7b2ffe9e994f8fa9932076a4adffc9f3126"}],"HostName":"d1e446a8b63c","MappedDirectories":[{"HostPath":"c:\\programdata\\iotedge\\mgmt","ContainerPath":"c:\\programdata\\iotedge\\mgmt","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":false}],"HvPartition":false,"EndpointList":["30ADC2FD-A576-4E70-915B-47A8168400C2"],"AllowUnqualifiedDNSQuery":true}).
√ latest security daemon - OK
‼ host time is close to real time - Warning
    Could not query NTP server
× container time is close to host time - Error
    Could not query local time inside container
‼ DNS server - Warning
    Container engine is not configured with DNS server setting, which may impact connectivity to IoT Hub.
    Please see https://aka.ms/iotedge-prod-checklist-dns for best practices.
    You can ignore this warning if you are setting DNS server per module in the Edge deployment.
‼ production readiness: certificates - Warning
    The Edge device is using self-signed automatically-generated development certificates.
    They will expire in 89 days (at 2021-01-06 12:07:44 UTC) causing module-to-module and downstream device communication to fail on an active deployment.
    After the certs have expired, restarting the IoT Edge daemon will trigger it to generate new development certs.
    Please consider using production certificates instead. See https://aka.ms/iotedge-prod-checklist-certs for best practices.
√ production readiness: container engine - OK
‼ production readiness: logs policy - Warning
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.
‼ production readiness: Edge Agent's storage directory is persisted on the host filesystem - Warning
    The edgeAgent module is not configured to persist its C:\Windows\Temp\edgeAgent directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.
× production readiness: Edge Hub's storage directory is persisted on the host filesystem - Error
    Could not check current state of edgeHub container

Connectivity checks
-------------------
√ host can connect to and perform TLS handshake with IoT Hub AMQP port - OK
√ host can connect to and perform TLS handshake with IoT Hub HTTPS / WebSockets port - OK
√ host can connect to and perform TLS handshake with IoT Hub MQTT port - OK
× container on the IoT Edge module network can connect to IoT Hub AMQP port - Error
    Container on the nat network could not connect to IotixHub.azure-devices.net:5671
× container on the IoT Edge module network can connect to IoT Hub HTTPS / WebSockets port - Error
    Container on the nat network could not connect to IotixHub.azure-devices.net:443
× container on the IoT Edge module network can connect to IoT Hub MQTT port - Error
    Container on the nat network could not connect to IotixHub.azure-devices.net:8883

10 check(s) succeeded.
5 check(s) raised warnings. Re-run with --verbose for more details.
6 check(s) raised errors. Re-run with --verbose for more details.

```
</details>

### Device Information
* Host OS : Windows Server 2019 Datacenter 1809 (Build 17763.1457)
* Architecture : amd64
* Container OS : Windows containers

### Runtime Versions
* iotedged [run `iotedge version`]: 1.0.9.5 (c602ee699e3004dc50fe4c2a4ae30765079b4855)
* Edge Agent [image tag (e.g. 1.0.0)]: N/A
* Edge Hub [image tag (e.g. 1.0.0)]: N/A
* Docker/Moby [run `docker version`]: 3.0.5

Note: when using Windows containers on Windows, run `docker -H npipe:////./pipe/iotedge_moby_engine version` instead

## Logs

<details>
<summary>Get-IoTEdgeLog</summary>

```

08.10.2020 14:08:10 info: edgelet_docker::runtime -- Starting module edgeAgent...
08.10.2020 14:08:11 warn: edgelet_utils::logging -- Could not start module edgeAgent
08.10.2020 14:08:11 warn: edgelet_utils::logging --     caused by: Could not start module edgeAgent                     
08.10.2020 14:08:11 warn: edgelet_utils::logging --     caused by: hcsshim::CreateComputeSystem                         
                    21ee2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8faf541ea698b03: Insufficient system resources
                    exist to complete the requested service.
                    (extra info: {"SystemType":"Container","Name":"21ee2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8fa
                    f541ea698b03","Owner":"docker","VolumePath":"\\\\?\\Volume{b07fffca-8b65-4391-a37b-0004ee6db30d}","
                    IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\21ee
                    2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8faf541ea698b03","Layers":[{"ID":"924cea96-5f03-52ed-9
                    050-4aa1631143db","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\a703a1603fbff4546871b09afd7
                    b4a1168a58df188f3eb8f72977e2d3131fdef"},{"ID":"1e985a6b-7d60-5f82-a0bb-f30e88879f2c","Path":"C:\\Pr
                    ogramData\\iotedge-moby\\windowsfilter\\f0a08218a7bffd07fb0d8028573cd430571b19155da1b23d4813ec970df
                    179ef"},{"ID":"2cbc743d-ded4-5625-8364-fd89d00c8017","Path":"C:\\ProgramData\\iotedge-moby\\windows
                    filter\\04898633e6d09e33a292ee4fe7c427e5a9d24a21c36ee67c4dd1de873911bdef"},{"ID":"e78ac4fe-21da-5d4
                    2-b5fd-836fdca02131","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\25104a9e10faddb23d6d6a12
                    900bf35d49c9b3f85db9d2895f1a20d610d8e0bb"},{"ID":"11b203dc-4419-5ea3-99b6-38eb03538d8e","Path":"C:\
                    \ProgramData\\iotedge-moby\\windowsfilter\\206ee5b1d0b78124448b2808c1a63c0103c2e9276109658c37d520a7
                    c94e5a5d"},{"ID":"21a0e4f6-d055-584a-b09b-a14f525f1a73","Path":"C:\\ProgramData\\iotedge-moby\\wind
                    owsfilter\\020898b022a6158be99993ec42f41026e8785956dffc5d27e30e7f836e50bf9c"},{"ID":"64308b2d-a68c-
                    5e31-ae60-58e14ebe5acc","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\b278e92ba4349b4b0a44c
                    15855a011091bae7085d87f672a3eccc406c408a604"},{"ID":"a586d6c1-551a-57b9-b8fe-82e7255673d0","Path":"
                    C:\\ProgramData\\iotedge-moby\\windowsfilter\\b4576284b52c3ec58092cb324113cc38e144d1ed9058dc23d06be
                    5a1d6e1d79e"},{"ID":"10c57b51-ce54-56ed-bb2f-9326d2d5cde5","Path":"C:\\ProgramData\\iotedge-moby\\w
                    indowsfilter\\f21ffaddfe875329bf552b59847ba85ceb4fbd0f3fc9eb297359773ee043b03b"},{"ID":"05a4c235-81
                    44-5c3b-a2e2-b9783b53aab4","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\59972063c3466de6b7
                    f73802101c64bec2e747779252fcebbd581fbf603b5f5b"},{"ID":"d31cb17a-7b99-5cf9-985c-8b1638379b49","Path
                    ":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\a5bed0e10e0500a1c0c2d4bd298f385a41b6d952538f511e59
                    ed61dd241f8701"},{"ID":"4f896c30-a2ac-503e-8831-7b510923d562","Path":"C:\\ProgramData\\iotedge-moby
                    \\windowsfilter\\607ceb328789165f7c1f246b52f467079c492c08cf571a057b89069dccc83b0d"}],"HostName":"21
                    ee2f79eaa1","MappedDirectories":[{"HostPath":"c:\\programdata\\iotedge\\mgmt","ContainerPath":"c:\\
                    programdata\\iotedge\\mgmt","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityV
                    M":false},{"HostPath":"c:\\programdata\\iotedge\\workload","ContainerPath":"c:\\programdata\\iotedg
                    e\\workload","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":false}],"HvP
                    artition":false,"EndpointList":["FF79F95F-7856-48AC-AD53-468D2896F7DB"],"AllowUnqualifiedDNSQuery":
                    true})
08.10.2020 14:08:11 warn: edgelet_utils::logging -- A module runtime error occurred.
08.10.2020 14:08:11 warn: edgelet_utils::logging --     caused by: hcsshim::CreateComputeSystem                         
                    21ee2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8faf541ea698b03: Insufficient system resources
                    exist to complete the requested service.
                    (extra info: {"SystemType":"Container","Name":"21ee2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8fa
                    f541ea698b03","Owner":"docker","VolumePath":"\\\\?\\Volume{b07fffca-8b65-4391-a37b-0004ee6db30d}","
                    IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\21ee
                    2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8faf541ea698b03","Layers":[{"ID":"924cea96-5f03-52ed-9
                    050-4aa1631143db","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\a703a1603fbff4546871b09afd7
                    b4a1168a58df188f3eb8f72977e2d3131fdef"},{"ID":"1e985a6b-7d60-5f82-a0bb-f30e88879f2c","Path":"C:\\Pr
                    ogramData\\iotedge-moby\\windowsfilter\\f0a08218a7bffd07fb0d8028573cd430571b19155da1b23d4813ec970df
                    179ef"},{"ID":"2cbc743d-ded4-5625-8364-fd89d00c8017","Path":"C:\\ProgramData\\iotedge-moby\\windows
                    filter\\04898633e6d09e33a292ee4fe7c427e5a9d24a21c36ee67c4dd1de873911bdef"},{"ID":"e78ac4fe-21da-5d4
                    2-b5fd-836fdca02131","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\25104a9e10faddb23d6d6a12
                    900bf35d49c9b3f85db9d2895f1a20d610d8e0bb"},{"ID":"11b203dc-4419-5ea3-99b6-38eb03538d8e","Path":"C:\
                    \ProgramData\\iotedge-moby\\windowsfilter\\206ee5b1d0b78124448b2808c1a63c0103c2e9276109658c37d520a7
                    c94e5a5d"},{"ID":"21a0e4f6-d055-584a-b09b-a14f525f1a73","Path":"C:\\ProgramData\\iotedge-moby\\wind
                    owsfilter\\020898b022a6158be99993ec42f41026e8785956dffc5d27e30e7f836e50bf9c"},{"ID":"64308b2d-a68c-
                    5e31-ae60-58e14ebe5acc","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\b278e92ba4349b4b0a44c
                    15855a011091bae7085d87f672a3eccc406c408a604"},{"ID":"a586d6c1-551a-57b9-b8fe-82e7255673d0","Path":"
                    C:\\ProgramData\\iotedge-moby\\windowsfilter\\b4576284b52c3ec58092cb324113cc38e144d1ed9058dc23d06be
                    5a1d6e1d79e"},{"ID":"10c57b51-ce54-56ed-bb2f-9326d2d5cde5","Path":"C:\\ProgramData\\iotedge-moby\\w
                    indowsfilter\\f21ffaddfe875329bf552b59847ba85ceb4fbd0f3fc9eb297359773ee043b03b"},{"ID":"05a4c235-81
                    44-5c3b-a2e2-b9783b53aab4","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\59972063c3466de6b7
                    f73802101c64bec2e747779252fcebbd581fbf603b5f5b"},{"ID":"d31cb17a-7b99-5cf9-985c-8b1638379b49","Path
                    ":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\a5bed0e10e0500a1c0c2d4bd298f385a41b6d952538f511e59
                    ed61dd241f8701"},{"ID":"4f896c30-a2ac-503e-8831-7b510923d562","Path":"C:\\ProgramData\\iotedge-moby
                    \\windowsfilter\\607ceb328789165f7c1f246b52f467079c492c08cf571a057b89069dccc83b0d"}],"HostName":"21
                    ee2f79eaa1","MappedDirectories":[{"HostPath":"c:\\programdata\\iotedge\\mgmt","ContainerPath":"c:\\
                    programdata\\iotedge\\mgmt","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityV
                    M":false},{"HostPath":"c:\\programdata\\iotedge\\workload","ContainerPath":"c:\\programdata\\iotedg
                    e\\workload","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":false}],"HvP
                    artition":false,"EndpointList":["FF79F95F-7856-48AC-AD53-468D2896F7DB"],"AllowUnqualifiedDNSQuery":
                    true})
08.10.2020 14:08:11 warn: edgelet_core::watchdog -- Error in watchdog when checking for edge runtime status:
08.10.2020 14:08:45 info: edgelet_core::watchdog -- Checking edge runtime status
08.10.2020 14:08:45 info: edgelet_docker::runtime -- Starting module edgeAgent...
08.10.2020 14:08:45 info: edgelet_core::watchdog -- Edge runtime status is stopped, starting module now...
08.10.2020 14:08:46 warn: edgelet_core::watchdog -- Error in watchdog when checking for edge runtime status:
08.10.2020 14:08:46 warn: edgelet_utils::logging -- A module runtime error occurred.
08.10.2020 14:08:46 warn: edgelet_utils::logging -- Could not start module edgeAgent
08.10.2020 14:08:46 warn: edgelet_utils::logging --     caused by: hcsshim::CreateComputeSystem                         
                    21ee2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8faf541ea698b03: Insufficient system resources
                    exist to complete the requested service.
                    (extra info: {"SystemType":"Container","Name":"21ee2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8fa
                    f541ea698b03","Owner":"docker","VolumePath":"\\\\?\\Volume{b07fffca-8b65-4391-a37b-0004ee6db30d}","
                    IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\21ee
                    2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8faf541ea698b03","Layers":[{"ID":"924cea96-5f03-52ed-9
                    050-4aa1631143db","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\a703a1603fbff4546871b09afd7
                    b4a1168a58df188f3eb8f72977e2d3131fdef"},{"ID":"1e985a6b-7d60-5f82-a0bb-f30e88879f2c","Path":"C:\\Pr
                    ogramData\\iotedge-moby\\windowsfilter\\f0a08218a7bffd07fb0d8028573cd430571b19155da1b23d4813ec970df
                    179ef"},{"ID":"2cbc743d-ded4-5625-8364-fd89d00c8017","Path":"C:\\ProgramData\\iotedge-moby\\windows
                    filter\\04898633e6d09e33a292ee4fe7c427e5a9d24a21c36ee67c4dd1de873911bdef"},{"ID":"e78ac4fe-21da-5d4
                    2-b5fd-836fdca02131","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\25104a9e10faddb23d6d6a12
                    900bf35d49c9b3f85db9d2895f1a20d610d8e0bb"},{"ID":"11b203dc-4419-5ea3-99b6-38eb03538d8e","Path":"C:\
                    \ProgramData\\iotedge-moby\\windowsfilter\\206ee5b1d0b78124448b2808c1a63c0103c2e9276109658c37d520a7
                    c94e5a5d"},{"ID":"21a0e4f6-d055-584a-b09b-a14f525f1a73","Path":"C:\\ProgramData\\iotedge-moby\\wind
                    owsfilter\\020898b022a6158be99993ec42f41026e8785956dffc5d27e30e7f836e50bf9c"},{"ID":"64308b2d-a68c-
                    5e31-ae60-58e14ebe5acc","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\b278e92ba4349b4b0a44c
                    15855a011091bae7085d87f672a3eccc406c408a604"},{"ID":"a586d6c1-551a-57b9-b8fe-82e7255673d0","Path":"
                    C:\\ProgramData\\iotedge-moby\\windowsfilter\\b4576284b52c3ec58092cb324113cc38e144d1ed9058dc23d06be
                    5a1d6e1d79e"},{"ID":"10c57b51-ce54-56ed-bb2f-9326d2d5cde5","Path":"C:\\ProgramData\\iotedge-moby\\w
                    indowsfilter\\f21ffaddfe875329bf552b59847ba85ceb4fbd0f3fc9eb297359773ee043b03b"},{"ID":"05a4c235-81
                    44-5c3b-a2e2-b9783b53aab4","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\59972063c3466de6b7
                    f73802101c64bec2e747779252fcebbd581fbf603b5f5b"},{"ID":"d31cb17a-7b99-5cf9-985c-8b1638379b49","Path
                    ":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\a5bed0e10e0500a1c0c2d4bd298f385a41b6d952538f511e59
                    ed61dd241f8701"},{"ID":"4f896c30-a2ac-503e-8831-7b510923d562","Path":"C:\\ProgramData\\iotedge-moby
                    \\windowsfilter\\607ceb328789165f7c1f246b52f467079c492c08cf571a057b89069dccc83b0d"}],"HostName":"21
                    ee2f79eaa1","MappedDirectories":[{"HostPath":"c:\\programdata\\iotedge\\mgmt","ContainerPath":"c:\\
                    programdata\\iotedge\\mgmt","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityV
                    M":false},{"HostPath":"c:\\programdata\\iotedge\\workload","ContainerPath":"c:\\programdata\\iotedg
                    e\\workload","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":false}],"HvP
                    artition":false,"EndpointList":["4E274386-009E-4E2B-A05F-0597D3B875BF"],"AllowUnqualifiedDNSQuery":
                    true})
08.10.2020 14:08:46 warn: edgelet_utils::logging --     caused by: hcsshim::CreateComputeSystem                         
                    21ee2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8faf541ea698b03: Insufficient system resources
                    exist to complete the requested service.
                    (extra info: {"SystemType":"Container","Name":"21ee2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8fa
                    f541ea698b03","Owner":"docker","VolumePath":"\\\\?\\Volume{b07fffca-8b65-4391-a37b-0004ee6db30d}","
                    IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\21ee
                    2f79eaa11577896c21663b6d49d4b3250d6a037e3bbbf8faf541ea698b03","Layers":[{"ID":"924cea96-5f03-52ed-9
                    050-4aa1631143db","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\a703a1603fbff4546871b09afd7
                    b4a1168a58df188f3eb8f72977e2d3131fdef"},{"ID":"1e985a6b-7d60-5f82-a0bb-f30e88879f2c","Path":"C:\\Pr
                    ogramData\\iotedge-moby\\windowsfilter\\f0a08218a7bffd07fb0d8028573cd430571b19155da1b23d4813ec970df
                    179ef"},{"ID":"2cbc743d-ded4-5625-8364-fd89d00c8017","Path":"C:\\ProgramData\\iotedge-moby\\windows
                    filter\\04898633e6d09e33a292ee4fe7c427e5a9d24a21c36ee67c4dd1de873911bdef"},{"ID":"e78ac4fe-21da-5d4
                    2-b5fd-836fdca02131","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\25104a9e10faddb23d6d6a12
                    900bf35d49c9b3f85db9d2895f1a20d610d8e0bb"},{"ID":"11b203dc-4419-5ea3-99b6-38eb03538d8e","Path":"C:\
                    \ProgramData\\iotedge-moby\\windowsfilter\\206ee5b1d0b78124448b2808c1a63c0103c2e9276109658c37d520a7
                    c94e5a5d"},{"ID":"21a0e4f6-d055-584a-b09b-a14f525f1a73","Path":"C:\\ProgramData\\iotedge-moby\\wind
                    owsfilter\\020898b022a6158be99993ec42f41026e8785956dffc5d27e30e7f836e50bf9c"},{"ID":"64308b2d-a68c-
                    5e31-ae60-58e14ebe5acc","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\b278e92ba4349b4b0a44c
                    15855a011091bae7085d87f672a3eccc406c408a604"},{"ID":"a586d6c1-551a-57b9-b8fe-82e7255673d0","Path":"
                    C:\\ProgramData\\iotedge-moby\\windowsfilter\\b4576284b52c3ec58092cb324113cc38e144d1ed9058dc23d06be
                    5a1d6e1d79e"},{"ID":"10c57b51-ce54-56ed-bb2f-9326d2d5cde5","Path":"C:\\ProgramData\\iotedge-moby\\w
                    indowsfilter\\f21ffaddfe875329bf552b59847ba85ceb4fbd0f3fc9eb297359773ee043b03b"},{"ID":"05a4c235-81
                    44-5c3b-a2e2-b9783b53aab4","Path":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\59972063c3466de6b7
                    f73802101c64bec2e747779252fcebbd581fbf603b5f5b"},{"ID":"d31cb17a-7b99-5cf9-985c-8b1638379b49","Path
                    ":"C:\\ProgramData\\iotedge-moby\\windowsfilter\\a5bed0e10e0500a1c0c2d4bd298f385a41b6d952538f511e59
                    ed61dd241f8701"},{"ID":"4f896c30-a2ac-503e-8831-7b510923d562","Path":"C:\\ProgramData\\iotedge-moby
                    \\windowsfilter\\607ceb328789165f7c1f246b52f467079c492c08cf571a057b89069dccc83b0d"}],"HostName":"21
                    ee2f79eaa1","MappedDirectories":[{"HostPath":"c:\\programdata\\iotedge\\mgmt","ContainerPath":"c:\\
                    programdata\\iotedge\\mgmt","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityV
                    M":false},{"HostPath":"c:\\programdata\\iotedge\\workload","ContainerPath":"c:\\programdata\\iotedg
                    e\\workload","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":false}],"HvP
                    artition":false,"EndpointList":["4E274386-009E-4E2B-A05F-0597D3B875BF"],"AllowUnqualifiedDNSQuery":
                    true})
08.10.2020 14:08:46 warn: edgelet_utils::logging --     caused by: Could not start module edgeAgent

```
</details>

<details>
<summary>edge-agent logs</summary>

```

N/A (Edge agent cannot start)

```
</details>

<details>
<summary>edge-hub logs</summary>

```

N/A (Edge agent cannot start)

```
</details>

## Additional Information

We have this issue on two different servers, both with the same Windows version, both virtualized.

Looking at the error, it looks like it's linked to this update https://support.microsoft.com/en-us/help/4570006/update-to-disable-and-remove-the-remotefx-vgpu-component which has disabled the Remotefx vgpu component. In the KB description they warn that trying to run a VM which has the RemoteFX option activated will throw this error.

**Comment by veyalla** [comment]
Since the issue seems unrelated to IoT Edge, I suggest posting this on the Windows containers forum https://social.msdn.microsoft.com/Forums/en-US/home?forum=windowscontainers

**Comment by Gimly** [comment]
@veyalla What makes you think this is unrelated to IoT Edge? It's the edge agent that can't start with a default installation on those machines.

**Comment by Gimly** [comment]
@veyalla I have found a temporary solution from here: https://github.com/Microsoft/hcsshim/issues/437. The fix was to set the registry `HKLM\SYSTEM\ControlSet001\Services\NDIS\IfTypes\24\IfUsedNetLuidIndices` to 01 (it was the max value) and reboot the computer.

The main cause seems to be a bug in hcsshim used by Moby in which a restart didn't release this registry key. But what is annoying is that the fix at hcsshim has been patched since june 2019 (looking at when the PR has been merged). 

Is IoT Edge using an earlier version of Moby on which the patch hasn't been applied?
